src/libsyntax/parse/parser.rs

```diff
@@ -5997,21 +5996,22 @@ impl<'a> Parser<'a> {
         let lo = self.span;
         let visibility = self.parse_visibility(false)?;
 
-        if self.check_keyword(keywords::Static) {
-            // FOREIGN STATIC ITEM
+        // FOREIGN STATIC ITEM
+        // Treat `const` as `static` for error recovery, but don't add it to expected tokens.
+        if self.check_keyword(keywords::Static) || self.token.is_keyword(keywords::Const) {
+            if self.token.is_keyword(keywords::Const) {
+                self.diagnostic()
+                    .struct_span_err(self.span, "extern items cannot be `const`")
+                    .span_label(self.span, "use `static` instead").emit();
```

Comment:
This could be `.span_suggestion(self.span, "instead try using", "static".to_owned())`, which would allow IDEs and tools to automatically replace it.